teps/0035-doc-policy-authentication-authorization-rbac.md

```diff
@@ -0,0 +1,635 @@
+in charge of `PodSecurityPolicy` has noted that shortcoming as a failing both at various KubeCon's and in the still
+[active KEP](https://github.com/kubernetes/enhancements/issues/5) around `PodSecurityPolicy`.
+
+And certainly use of for example any OPA integrations can be employ from the start whenever a new cluster is created if such an
+approache is tenable.
+
+So for now, as we consider this for Tekton, we are focused on facilitating easy migration to more security.
+
+### User Experience (optional)
+
+The documentation we are looking to provide stems from the motivation to assist the user experience in the
+security/policy/authentication/authorization relam.
+
+### Performance (optional)
+
+Some points of note with OPA Gatekeeper's implementation and performance:
+
+- it injects itself into the Kubernetes cluster as and admission webhook
+- the `match` section of the `spec` in the `Constraint` OPA CRD serves as a filter for which API objects the webhook deals with
+- Gatekeeper employs the Kubernetes controller / watch / informer constructs for providing that data requested in 
+`parameters` section of the `spec` in the `Constraint` OPA CRD.  What that means is the latest versions of those objects
+are obtained in the background, and then when the `ConstraintTemplate` is executed, instead of explicit calls to the
+API server to retrieve those objects, the controller cache is leveraged.
+
+Between the `match` filtering and use of a controller cache, and that general admission webhook overhead is generally accepted
+in the Kubernetes ecosystem as more than tolerable, the only overhead of note will be OPA's Rego processing engine.
+
+
+
+## Design Details
+
+At this time, with documentations and possibly references to policies hosted outside of https://github.com/tektoncd
+there is no coding design within Tekton to declare.
+
+We are building upon the fact that prior K8s design choices that allow for plugging in such solutions (i.e. admission
+webhooks) is the key design point here.
+ 
+## Test Plan
+
+For now, any testing of third party catalogs of existing policies would at least initially be the providence of 
+those third party sources.
+
+That said, if our comfort level for:
+
+- the stability of the third party reference policy
+- the fact that it can be obtained by a `wget` HTTP GET just as easily as Tekton's various K8s yaml config files
+- and there is a containable means of installing said third party policy engines in our existing e2e's
+
+We can eventually see about validating the scenarios discussed here in Tekton CI.
+
+## Drawbacks
+
+Having to install a separate components like Gatekeeper into a cluster vs. having solutions land with just the 
+installing Pipelines may displease some consumers.  We'll have to track community response as things progress and
+adjust if needed.  But so far, there has been more acceptance than pushback.
```

Comment:
i was discussing with someone recently (maybe @afrittoli ?) that it would be really need to have an example repo  (e.g. something like tektoncd/example) where we have triggers + pipelines setup and folks can pretty much copy it to get up and running - maybe if we have that one day we could include gatekeeper in the installation

Reply:
Interesting notion ... yeah let's try to collectively remember to include policy engine based stuff (opa, gatekeeper, or if other vendors happen to gain some traction by then) if such a thing came to fruition.